app/models/lookup_value.rb

```diff
@@ -28,7 +28,7 @@ class LookupValue < ActiveRecord::Base
   scoped_search :in => :lookup_key, :on => :key, :rename => :lookup_key, :complete_value => true
 
   def value_present?
-    self.errors.add(:value, :blank) if value.nil? && !use_puppet_default
+    self.errors.add(:value, :blank) if value.to_s.empty? && !use_puppet_default && lookup_key.puppet?
```

Comment:
@dLobatog I opened: http://projects.theforeman.org/issues/12822.
I don't think it should be part of this PR because there might be more puppet specific behavior there and I'd prefer moving it all together and not forgetting some leftovers (this PR fixes leftovers from the STI change).